x/gov/keeper/tally.go

```diff
@@ -40,25 +40,25 @@ func (keeper Keeper) Tally(ctx sdk.Context, proposal types.Proposal) (passes boo
 		if val, ok := currValidators[valAddrStr]; ok {
 			val.Vote = vote.Option
 			currValidators[valAddrStr] = val
-		} else {
-			// iterate over all delegations from voter, deduct from any delegated-to validators
-			keeper.sk.IterateDelegations(ctx, vote.Voter, func(index int64, delegation exported.DelegationI) (stop bool) {
-				valAddrStr := delegation.GetValidatorAddr().String()
+		}
 
-				if val, ok := currValidators[valAddrStr]; ok {
-					val.DelegatorDeductions = val.DelegatorDeductions.Add(delegation.GetShares())
-					currValidators[valAddrStr] = val
+		// iterate over all delegations from voter, deduct from any delegated-to validators
+		keeper.sk.IterateDelegations(ctx, vote.Voter, func(index int64, delegation exported.DelegationI) (stop bool) {
+			valAddrStr := delegation.GetValidatorAddr().String()
 
-					delegatorShare := delegation.GetShares().Quo(val.DelegatorShares)
-					votingPower := delegatorShare.MulInt(val.BondedTokens)
+			if val, ok := currValidators[valAddrStr]; ok {
```

Comment:
So, this now introduces a new problem of the self-delegations deducting delegator amounts from their own validator! This can be resolved by adding the following check: 
```
@ line 39 
  voteValAddrStr := sdk.ValAddress(vote.Voter).String()
@ line 49:
  val, ok := currValidators[valAddrStr]
   if ok && valAddrStr != voteValAddrStr {
```

Reply:
Nice suggestion!